### PR TITLE
Separate spec from schema

### PR DESCRIPTION
**Pull request by jeromekelleher**
Here's an initial pass at #46. After much messing around, I think the only acceptable way to do this is to separate out the schema from the specification, and to take all the narrative documentation out of the JSON Schema definition. In principle it's great idea to have a single file that defines everything, but it's just to messy to try to embed restructured text into the JSON schema, and the automatic rendering of documentation we're currently getting from sphinx-jsonschema just isn't up to something this complicated.

So, I suggest we separate the concerns by making the schema definitive structurally and in terms of types, and to make the markdown text definitive in terms of the actual human readable specification. To avoid duplication, I've taken all the narrative elements out of the JSON schema.

FWIW, I think this makes the schema itself much more readable, as it's not cluttered up with a bunch of rst.

@grahamgower, what do you think? I'm running this by you before I go much further, as I realise you put a lot of work into this, so may not be a big fan!

### CODE REVIEW

**Comment by grahamgower** [comment]
I agree that this is the way forward. Using sphinx-jsonschema was an experiment that unfortunately didn't pay off.

**Comment by jeromekelleher** [comment]
Great, thanks @grahamgower, glad you said that!

I've done the basic moving around of content here without changing very much. It would be good to get this much merged, as I'd like to tackle the bigger structural issues now (lack of description of the population genetics model, and separating the fully qualified spec from the "user" spec here, and describing the role of the parser.)

File: docs/specification.md [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+``(start_time, end_time]``, and it may continue to exist after
+contributing ancestry to a descendant deme. The deme's ``end_time`` is
+defined as the ``end_time`` of the deme's last epoch.
+
+
+name
+: A string identifier for a deme, which MUST be unique
+  among all demes in a model.
+  Must be a valid
+  [python identifier](https://docs.python.org/3/reference/lexical_analysis.html#identifiers>)
```
Comment (grahamgower) [inline docs/specification.md:111]: Looks like a stray `>` at the end of the url.

File: docs/specification.md [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+     and the deme's ``start_time`` is not specified.
+   - If the deme has one ancestor with an ``end_time == 0``,
+     and the deme's ``start_time`` is not specified.
+   - If the deme has zero ancestors and a finite ``start_time``.
+
+epochs:
+: The list of {ref}`epochs <sec_spec_defs_epoch>` for this deme.
+
+defaults:
+: The default values for omitted properties in epochs.
```
Comment (grahamgower) [inline docs/specification.md:168]: Stray `:` after `epochs:` and `defaults:`.

**Review by grahamgower — APPROVED** [review]
LGTM!

**Pull request merged**